src/libs/ajax/data-table-providers/EntityServiceDataTableProvider.ts

```diff
@@ -41,4 +55,32 @@ export class EntityServiceDataTableProvider implements DataTableProvider {
   downloadTsv = (signal: AbortSignal, entityType: string): Promise<Blob> => {
     return Ajax(signal).Workspaces.workspace(this.namespace, this.name).getEntitiesTsv(entityType)
   }
+
+  isInvalid: isInvalidFn = (modeMatches: boolean, filePresent: boolean, match: boolean, _: boolean) => {
+    return modeMatches && filePresent && match
+  }
+
+  disabled: disabledFn = (filePresent: boolean, isInvalid: boolean, uploading: boolean, _: boolean) => {
+    return !filePresent || isInvalid || uploading
+  }
+
+  tooltip: tooltipFn = (filePresent: boolean, isInvalid: boolean, _: boolean) => {
+    return !filePresent || isInvalid ? 'Please select valid data to upload' : 'Upload selected data'
+  }
+
+  doUpload: uploadFn = async (uploadParams: UploadParameters) => {
+    const workspace = Ajax().Workspaces.workspace(uploadParams.namespace, uploadParams.name)
+    if (uploadParams.useFireCloudDataModel) {
+      return workspace.importEntitiesFile(uploadParams.file, { deleteEmptyValues: uploadParams.deleteEmptyValues })
+    } else {
+      const filesize = uploadParams.file?.size || Number.MAX_SAFE_INTEGER
+      if (filesize < 524288) { // 512k
+        return workspace.importFlexibleEntitiesFileSynchronous(uploadParams.file, { deleteEmptyValues: uploadParams.deleteEmptyValues })
+      } else {
+        const { jobId } = await workspace.importFlexibleEntitiesFileAsync(uploadParams.file, { deleteEmptyValues: uploadParams.deleteEmptyValues })
+        asyncImportJobStore.update(Utils.append({ targetWorkspace: { namespace: uploadParams.namespace, name: uploadParams.name }, jobId }))
+        notifyDataImportProgress(jobId)
+      }
+    }
```

Comment:
NOT in scope for this PR, but just for the exposure, making sure you're aware of items in the backlog: https://broadworkbench.atlassian.net/browse/AJ-405